l1-contracts/contracts/bridge/L1AssetRouter.sol

```diff
@@ -537,463 +279,104 @@ contract L1AssetRouter is IL1AssetRouter, ReentrancyGuard, Ownable2StepUpgradeab
-    /// @param _l2TxGasPerPubdataByte The gasPerPubdataByteLimit to be used in the corresponding L2 transaction.
-    /// @param _refundRecipient The address on L2 that will receive the refund for the transaction.
-    /// @dev If the L2 deposit finalization transaction fails, the `_refundRecipient` will receive the `_l2Value`.
-    /// Please note, the contract may change the refund recipient's address to eliminate sending funds to addresses
-    /// out of control.
-    /// - If `_refundRecipient` is a contract on L1, the refund will be sent to the aliased `_refundRecipient`.
-    /// - If `_refundRecipient` is set to `address(0)` and the sender has NO deployed bytecode on L1, the refund will
-    /// be sent to the `msg.sender` address.
-    /// - If `_refundRecipient` is set to `address(0)` and the sender has deployed bytecode on L1, the refund will be
-    /// sent to the aliased `msg.sender` address.
-    /// @dev The address aliasing of L1 contracts as refund recipient on L2 is necessary to guarantee that the funds
-    /// are controllable through the Mailbox, since the Mailbox applies address aliasing to the from address for the
-    /// L2 tx if the L1 msg.sender is a contract. Without address aliasing for L1 contracts as refund recipients they
-    /// would not be able to make proper L2 tx requests through the Mailbox to use or withdraw the funds from L2, and
-    /// the funds would be lost.
-    /// @return l2TxHash The L2 transaction hash of deposit finalization.
-    function depositLegacyErc20Bridge(
-        address _prevMsgSender,
-        address _l2Receiver,
-        address _l1Token,
-        uint256 _amount,
-        uint256 _l2TxGasLimit,
-        uint256 _l2TxGasPerPubdataByte,
-        address _refundRecipient
-    ) external payable override onlyLegacyBridge nonReentrant whenNotPaused returns (bytes32 l2TxHash) {
-        require(_l1Token != L1_WETH_TOKEN, "ShB: WETH deposit not supported 2");
-
-        bytes32 _assetId;
-        bytes memory l2BridgeMintCalldata;
-
-        {
-            // Inner call to encode data to decrease local var numbers
-            _assetId = _ensureTokenRegisteredWithNTV(_l1Token);
-
-            // solhint-disable-next-line func-named-parameters
-            l2BridgeMintCalldata = abi.encode(
-                _amount,
-                _prevMsgSender,
-                _l2Receiver,
-                getERC20Getters(_l1Token),
-                _l1Token
+    /// @notice Generates a calldata for calling the deposit finalization on the L2 native token contract.
+    /// @param _chainId The chain ID of the ZK chain to which deposit.
+    /// @param _l1Sender The address of the deposit initiator.
+    /// @param _assetId The deposited asset ID.
+    /// @param _transferData The encoded data, which is used by the asset handler to determine L2 recipient and amount. Might include extra information.
+    /// @return Returns calldata used on ZK chain.
+    function _getDepositL2Calldata(
```

Comment:
Unfortunately we might want to have this automatic token bridging feature for L2->L1 and in the future L2->L2 bridging as well, so it might need to be moved to the base

Reply:
Ok sounds good, I only a have a few minor things to resolve and want to double check circular inheritance